tests/unit/sidecars/live_data_watcher_loader_tests.py

```diff
@@ -0,0 +1,104 @@
+import os
+from unittest import mock
+import pytest
+import json
+
+import boto3
+from botocore.client import ClientError
+from moto import mock_s3
+
+from baseplate.sidecars.live_data_watcher import NodeWatcher
+from baseplate.sidecars.live_data_watcher import _load_from_s3
+from baseplate.sidecars.live_data_watcher import _parse_loader_type
+from baseplate.sidecars.live_data_watcher import LoaderType
+
+@pytest.fixture()
+def start_mock_s3():
+    with mock_s3():
+        yield
+
+@pytest.mark.parametrize('data,return_value', (
+    # Non-RAW configs
+    (json.dumps({'live_data_watcher_load_type': 'S3'}).encode('utf-8'), LoaderType.S3),
+
+    # Everything else should be RAW
+    (b'', LoaderType.RAW),
+    (json.dumps({'key': 'value'}).encode('utf-8'), LoaderType.RAW),
+    (json.dumps([]).encode('utf-8'), LoaderType.RAW),
+    (json.dumps(1234).encode('utf-8'), LoaderType.RAW),
+    (json.dumps(None).encode('utf-8'), LoaderType.RAW),
+    # TODO: fuzzing
+))
+def test_parse_loader_type(data, return_value):
+    assert _parse_loader_type(data) == return_value
+
+
+@pytest.mark.parametrize("missing_config", (
+    "region_name",
+    "bucket_name",
+    "file_key",
+    "sse_key",
+))
+def test_load_from_s3_missing_config(start_mock_s3, missing_config):
+    data = {
+        "region_name": "some_region",
+        "bucket_name": "some_bucket",
+        "file_key": "some_file",
+        "sse_key": "some_key",
+    }
+    data.pop(missing_config)
+
+    assert _load_from_s3(json.dumps(data).encode('utf-8')) is None
+
+
+def test_successful_load_from_s3(start_mock_s3):
+    bucket = 'my-test-bucket'
+    key = 'my-test-key'
+    contents = b'my-test-contents'
+
+    s3 = boto3.resource('s3')
+    s3.Bucket(bucket).create()
+    obj = s3.Object(bucket, key)
+    obj.put(Body=contents)
+
+    data = _load_from_s3(json.dumps({
+        "region_name": "us-east-1",
+        "bucket_name": bucket,
+        "file_key": key,
+        "sse_key": "key"
+    }).encode('utf-8'))
+
+    assert data == contents
+
+
+@pytest.mark.parametrize('exc', (
+    ValueError,
+    ClientError({'Error': {'Code': 'error', 'Message': 'msg'}}, 'operation'),
+))
+def test_unsuccessful_load_from_s3(start_mock_s3, exc):
+    with mock.patch('baseplate.sidecars.live_data_watcher.boto3.client') as client:
```

Comment:
This use of a mock.MagicMock limits what you're able to test here. Did you consider using [`botocore.stub.Stubber`](https://botocore.amazonaws.com/v1/documentation/api/latest/reference/stubber.html) instead?

Reply:
TIL about Stubber, I was looking at moto and couldn't find anything in there so fell back to mock